crates/re_viewer/src/ui/selection_panel.rs

```diff
@@ -553,11 +555,232 @@ fn space_view_top_level_properties(
                         .class(ctx.space_view_class_registry)
                         .display_name(),
                 );
+
                 ui.end_row();
             });
     }
 }
 
+/// State of the space origin widget.
+#[derive(Default, Clone)]
+enum SpaceOriginEditState {
+    #[default]
+    NotEditing,
+
+    Editing {
+        /// The string currently entered by the user.
+        origin_string: String,
+
+        /// Did we just enter editing mode?
+        entered_editing: bool,
+
+        /// The index of the currently selected suggestion (for keyboard navigation).
+        selected_suggestion: Option<usize>,
+    },
+}
+
+/// Display the space origin of a space view.
+fn space_view_space_origin_widget_ui(
+    ui: &mut Ui,
+    ctx: &ViewerContext<'_>,
+    spaces_info: &SpaceInfoCollection,
+    space_view: &SpaceViewBlueprint,
+) {
+    let is_editing_id = ui.make_persistent_id(space_view.id.hash());
+    let mut state: SpaceOriginEditState =
+        ui.memory_mut(|mem| mem.data.get_temp(is_editing_id).unwrap_or_default());
+
+    match &mut state {
+        SpaceOriginEditState::NotEditing => {
+            let mut space_origin_string = space_view.space_origin.to_string();
+            let output = egui::TextEdit::singleline(&mut space_origin_string).show(ui);
+
+            if output.response.gained_focus() {
+                state = SpaceOriginEditState::Editing {
+                    origin_string: space_origin_string,
+                    entered_editing: true,
+                    selected_suggestion: None,
+                };
+            }
+        }
+        SpaceOriginEditState::Editing {
+            origin_string,
+            entered_editing,
+            selected_suggestion,
+        } => {
+            let keep_editing = space_view_space_origin_widget_editing_ui(
+                ui,
+                ctx,
+                spaces_info,
+                origin_string,
+                *entered_editing,
+                space_view,
+                selected_suggestion,
+            );
+
+            if keep_editing {
+                *entered_editing = false;
+            } else {
+                state = SpaceOriginEditState::NotEditing;
+            }
+        }
+    }
+
+    ui.memory_mut(|mem| mem.data.insert_temp(is_editing_id, state));
+}
+
+/// Display the space origin of a space view with it is in edit mode.
+fn space_view_space_origin_widget_editing_ui(
+    ui: &mut Ui,
+    ctx: &ViewerContext<'_>,
+    spaces_info: &SpaceInfoCollection,
+    space_origin_string: &mut String,
+    entered_editing: bool,
+    space_view: &SpaceViewBlueprint,
+    selected_suggestion: &mut Option<usize>,
+) -> bool {
+    let mut keep_editing = true;
+
+    //
+    // Build and filter the suggestion lists
+    //
+
+    // All suggestions for this class of space views.
+    let space_view_suggestions =
```

Comment:
I'm a bit confused about why we care about the possible space views here. Isn't this an "arbitrary entity path" picker? I.e. we'd like do select any possible entity path between the root and all leaves (including any entity paths bwetween that nothing is logged to).
I think we shouldn't care whether it shows up in the (quite dubious) list of `all_possible_space_views`

As previously mentioned, I want to remove this method in all cases anyways. But if you want we can take this separately if you want

Reply:
in either case this should be a list of origins (paths), not `SpaceViewBlueprint`, that's the only thing we're intersted in here

Reply:
👉🏻 https://github.com/rerun-io/rerun/issues/4895